src/middlewared/middlewared/plugins/interface/configure.py

```diff
@@ -137,14 +137,18 @@ def configure(self, data, aliases, wait_dhcp=False, options=None):
                     self.logger.debug('{}: removing possible valid_lft and preferred_lft on {}'.format(name, addr))
                     iface.replace_address(addr)
 
-        # carp must be configured after removing addresses
-        # in case removing the address removes the carp
         if carp_vhid:
-            if not self.middleware.call_sync('system.is_freenas') and not advskew:
-                if self.middleware.call_sync('failover.node') == 'A':
+            if self.middleware.call_sync('failover.licensed') and not advskew:
+                if 'NO_FAILOVER' in self.middleware.call_sync('failover.disabled_reasons'):
+                    if self.middleware.call_sync('failover.vip.get_states')[0]:
+                        advskew = 20
+                    elif self.middleware.call_sync('failover.status') != 'MASTER':
```

Comment:
Should this just be an else? otherwise advskew may be undefined

Reply:
Yep, I'll fix.